homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -96,3 +113,134 @@ async def client_listen(
                 self.client_listen(hass, entry, automower_client),
                 "reconnect_task",
             )
+
+    def _async_add_remove_device(self) -> None:
+        """Add new device, remove non-existing device."""
+        current_device = set(self.data)
+
+        # Skip update if no changes
+        if current_device == self._device_last_update:
+            return
+
+        # Process removed device
+        removed_device = self._device_last_update - current_device
```

Comment:
```suggestion
        removed_devices = self._devices_last_update - current_devices
```

nitpicky, but given those are all sets with potentially multiple devices in them we should make that clear in the naming already